src/client_side.cc

```diff
@@ -1503,10 +1452,7 @@ bool ConnStateData::serveDelayedError(Http::Stream *context)
         repContext->setReplyToStoreEntry(sslServerBump->entry, "delayed SslBump error");
 
         // save the original request for logging purposes
-        if (!context->http->al->request) {
-            context->http->al->request = http->request;
-            HTTPMSGLOCK(context->http->al->request);
-        }
+        context->http->initAleRequest();
```

Comment:
> I would expect it to be saved already because initAleRequest() is called from initRequest().

I think that is true.

>  I wonder whether we are introducing a bug here and/or in similar initAleRequest() calls.

I do not think we introduce a bug here: that method is called when this ConnStateData::serveDelayedError() returns true (thus stopping further processing in processParsedRequest()/clientProcessRequest()). I think both these calls could be safely removed.
Other two uses of initAleRequest() is doCallouts() (with a warning message) and initRequest(). Probably we can eliminate this method at all, moving its code into initRequest().